src/components/Autocomplete/index.js

```diff
@@ -0,0 +1,124 @@
+};
+
+// We handle the query length validation here as we don‘t want to do it in the
+// (throttled) typeahead search. If we did, the first input after
+// MIN_QUERY_LENGTH would be delayed.
+const handleFetchRequest = (endpoint, e, typeaheadSearch, clearSuggestions) => {
+  let query = e.value;
+
+  if (query) {
+    query = query.trim().toLowerCase();
+
+    if (query.length > MIN_QUERY_LENGTH) {
+      return typeaheadSearch(endpoint, query);
+    }
+  }
+
+  return clearSuggestions();
+};
+
+const handleBlur = (displayedValue, displaySuggestion, updateSuggestionInputValue, helper) => {
+  const savedValue = helper.value && displaySuggestion(helper.value.data);
+
+  // If the saved value in the redux-form matches the displayed value in the
+  // auto-suggest input, we know that the user hasn‘t changed his choice.
+  // If it did, we must reset both values as they aren‘t valid anymore.
+  if (savedValue !== displayedValue) {
+    updateSuggestionInputValue('');
+    helper.onChange('');
+  }
+};
+
+const handleChange = (newValue, updateSuggestionInputValue, helper) => {
+  // We have to manually update the displayed value in the auto-suggest.
+  // This has nothing to do with the actual redux-form value.
+  updateSuggestionInputValue(newValue);
+
+  // If there is no more value, then we know the user has deleted his choice,
+  // and we can update the redux-form store.
+  if (!newValue) {
+    helper.onChange('');
+  }
+};
+
+const handleSelection = (e, { suggestion }, helper) => {
+  // Disable form submitting when selecting option with ENTER
+  e.preventDefault();
+
+  // Update redux-form store manually to reflect the selection
+  helper.onChange({
+    data: suggestion,
+    id: suggestion.id,
+    class: 'City'
+  });
+};
+
+const Autocomplete = ({
+  endpoint,
+  suggestionFormatter,
+  typeaheadSearch,
+  clearSuggestions,
+  updateSuggestionInputValue,
+  suggestions,
+  value,
+  helper,
+  name,
+  html
+}) => (
+  <Autosuggest
+    theme={{
+      containerOpen: styles.open,
+      input: getClassname('input', fieldIsInvalid(helper)),
+      suggestionsList: styles.list,
+      suggestion: styles.suggestion,
+      suggestionFocused: styles.focused
+    }}
+    id={name}
+
+    suggestions={suggestions}
+    renderSuggestion={suggestionFormatter}
+    getSuggestionValue={suggestionFormatter}
+    focusFirstSuggestion
+
+    onSuggestionsFetchRequested={(e) => (
+      handleFetchRequest(endpoint, e, typeaheadSearch, clearSuggestions)
+    )}
+    onSuggestionsClearRequested={clearSuggestions}
+    onSuggestionSelected={(e, option) => (
+      handleSelection(e, option, helper)
+    )}
+
+    inputProps={{
+      name: name,
+      id: name,
+      required: html.required,
+      placeholder: html.placeholder,
+      autoComplete: html.autocomplete || 'off',
+
+      ...domOnlyProps(helper),
+
+      value: value,
+      onBlur: () => (
+        handleBlur(value, suggestionFormatter, updateSuggestionInputValue, helper)
```

Comment:
Yes, I guess that it would probably look better with React.createClass. I‘ll try